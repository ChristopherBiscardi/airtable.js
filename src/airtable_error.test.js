import cases from "jest-in-case";
import AirtableError from "./airtable_error";

const data = [
  { err: "my-error", msg: "some message", code: 500 },
  { err: "my-error", msg: "some message", code: undefined }
];

describe("airtable_error", () => {
  cases(
    "new AirtableError(error, message, statusCode)",
    ({ err, msg, code }) => {
      expect(new AirtableError(err, msg, code)).toMatchSnapshot();
    },
    data
  );
  cases(
    "new AirtableError(error, message, statusCode)",
    ({ err, msg, code }) => {
      expect(new AirtableError(err, msg, code).toString()).toMatchSnapshot();
    },
    data
  );
});
