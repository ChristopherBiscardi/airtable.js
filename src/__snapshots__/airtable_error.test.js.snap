// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`airtable_error new AirtableError(error, message, statusCode) case: 1 1`] = `
AirtableError {
  "error": "my-error",
  "message": "some message",
  "statusCode": 500,
}
`;

exports[`airtable_error new AirtableError(error, message, statusCode) case: 1 2`] = `"some message(my-error)[Http code 500]"`;

exports[`airtable_error new AirtableError(error, message, statusCode) case: 2 1`] = `
AirtableError {
  "error": "my-error",
  "message": "some message",
  "statusCode": undefined,
}
`;

exports[`airtable_error new AirtableError(error, message, statusCode) case: 2 2`] = `"some message(my-error)"`;
